src/snowflake/snowpark/modin/plugin/_internal/frame.py

```diff
@@ -672,6 +715,90 @@ def validate_no_duplicated_data_columns_mapped_for_labels(
                 message += f" {user_frame_identifier}"
             raise ValueError(message)
 
+    def cached_data_column_snowpark_pandas_types(
```

Comment:
can be a cached property

Reply:
Unfortunately we can not cache a mutable object. This will break the immutability guarantee of InternalFrame.

Reply:
InternalFrame is immutable so we can use cached property right?

Reply:
We can only use it if return type is immutable. For example if return type is int/tuple we can cache it but for list/dict we can not cache it. 
Consider following scenario:
```
x = frame.cached_data_column_snowpark_pandas_types
x["abc"] = "xyz"
print(frame.cached_data_column_snowpark_pandas_types) # this will have abc -> xyz in output
```

Reply:
Oh, good lesson! How about return tuple instead of list.

Reply:
We could return tuple and have `cached_property`, but `InternalFrame.create()` takes a `list`, so I'd have to either change that argument to `Sequence`, or make  the places that construct `data_column_types` out of an existing `cached_data_column_snowpark_pandas_types` switch the type from tuple to list. The performance gains from `cached_property` are minimal so I think `property` is ok.